### ISSUE TITLE
skaffold v1.37.0 inserts skaffold.dev/run-id label in wrong place

### ISSUE THREAD

**Issue by yteraoka** [GoogleContainerTools/skaffold#7219]
### Expected behavior

`skaffold.dev/run-id` label is inserted in the correct place

### Actual behavior

Inserted in the wrong place in some resources

Example

```
apiVersion: compute.cnrm.cloud.google.com/v1beta1
kind: ComputeAddress
metadata:
  annotations:
    cnrm.cloud.google.com/deletion-policy: abandon
  labels:
    app.kubernetes.io/component: example
    app.kubernetes.io/instance: example
    app.kubernetes.io/managed-by: kustomize
    app.kubernetes.io/name: example
    app.kubernetes.io/part-of: example
    app.kubernetes.io/version: dev
    skaffold.dev/run-id: fcbaf2a6-4bb2-4d82-9848-f8ebabe1fddb
  name: example
  namespace: example
spec:
  labels:
    skaffold.dev/run-id: fcbaf2a6-4bb2-4d82-9848-f8ebabe1fddb
  location: global
```

This will result in the following error

```
error: error validating "STDIN": error validating data: ValidationError(ComputeAddress.spec): unknown field "labels" in com.google.cloud.cnrm.compute.v1beta1.ComputeAddress.spec; if you choose to ignore these errors, turn validation off with --validate=false
```

Another example

```
apiVersion: compute.cnrm.cloud.google.com/v1beta1
kind: ComputeForwardingRule
metadata:
  labels:
    app.kubernetes.io/component: example
    app.kubernetes.io/instance: example
    app.kubernetes.io/managed-by: kustomize
    app.kubernetes.io/name: example
    app.kubernetes.io/part-of: example
    app.kubernetes.io/version: dev
    skaffold.dev/run-id: fcbaf2a6-4bb2-4d82-9848-f8ebabe1fddb
  name: example
  namespace: example
spec:
  ipAddress:
    addressRef:
      name: example
  ipProtocol: TCP
  labels:
    skaffold.dev/run-id: fcbaf2a6-4bb2-4d82-9848-f8ebabe1fddb
  location: global
  portRange: "80"
  target:
    targetHTTPProxyRef:
      name: https-redirector
      namespace: cnrm-global
```

### Information

- Skaffold version: v1.37.0 (I had no problem with v1.36.0)
- Operating system: client: Linux on circle ci / macOS, Server GKE 1.21.9
- Installed via: skaffold.dev
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v2beta10
kind: Config
metadata:
  name: editor
build:
  artifacts:
  - image: ***
    docker:
      dockerfile: './Dockerfile.hosting'
  local:
    useBuildkit: true
deploy:
  statusCheckDeadlineSeconds: 300
  kustomize:
    paths: ['k8s/hosting/overlays/local']

profiles:
- name: grpc-server/dev
  build:
    artifacts:
    - image: ***
      kaniko:
        cache: {}
        dockerfile: './Dockerfile.grpcserver'
        buildArgs:
          ENV: dev
    - image: ***
      docker:
        dockerfile: './Dockerfile.grpcserver-with-envoy'
        buildArgs:
          ENV: dev
    googleCloudBuild:
      projectId: ***
      timeout: 2400s
      kanikoImage: gcr.io/kaniko-project/executor:v1.6.0
      machineType: E2_HIGHCPU_8
  deploy:
    kustomize:
      paths: ['k8s/grpc-server/overlays/dev']
```

**Comment by aaron-prindle** [comment]
This issue was introduced in the change here #7056.  I have submitted a fix here #7221.  In the meantime unfortunately you will have to use skaffold v1.36.0 as there is no simple way to resolve this in v1.37.0.

**Comment by yteraoka** [comment]
I see, thanks
